### PR TITLE
Filter errors when marking sync failures for metrics

### PR DESCRIPTION
**Pull request by swetharepakula**
The following errors will not be counted as failed syncs for PSC metrics:
  * GCE Resource Not Found
  * GCE Bad Gateway
  * Service Not Found in Store
  * FwdRule IP does not match Service

/assign @prameshj @freehan

### CODE REVIEW

File: pkg/psc/controller_test.go [hunk 1]
```diff
@@ -728,6 +729,70 @@ func TestShouldProcess(t *testing.T) {
...
+		err           error
+		expectedError error
+	}{
+		{
+			desc:          "google api status not found error",
+			err:           utils.FakeGoogleAPINotFoundErr(),
+			expectedError: nil,
+		},
+		{
+			desc:          "google api bad gateway error",
```
Comment (prameshj) [inline pkg/psc/controller_test.go:745]: Also add a case for google api with a different status code that you are not filtering - like 409(Conflict) or 400(Bad Request)

File: pkg/psc/controller.go [hunk 2]
```diff
@@ -638,3 +648,21 @@ func shouldProcess(old, cur *sav1beta1.ServiceAttachment) bool {
...
+
+// filterError filters out errors that should not be considered sync errors for metrics
+func filterError(err error) error {
+	var apiError *googleapi.Error
+	if errors.As(err, &apiError) {
+		if utils.IsHTTPErrorCode(apiError, http.StatusNotFound) || utils.IsHTTPErrorCode(apiError, http.StatusBadGateway) {
+			return nil
+		}
+	}
+
```
Comment (prameshj) [inline pkg/psc/controller.go:660]: nit: newline can be skipped here and in line 666.

**Review by prameshj — COMMENTED** [review]
lgtm except one testcase suggestion.

File: pkg/psc/controller_test.go (hunk 1 above)
Comment (swetharepakula) [reply pkg/psc/controller_test.go]: Did some rework on the test and code. Changed the code to filter out BadRequests and now test both BadRequests and BadGateways where I expect BadGateways to return an error.

File: pkg/psc/controller.go (hunk 2 above)
Comment (swetharepakula) [reply pkg/psc/controller.go]: Fixed

**Pull request merged**